pkg/scheduler/types.go

```diff
@@ -0,0 +1,43 @@
+package scheduler
+
+import (
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/graph"
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+// A Build is all the information required for a build
+type Build struct {
+	Spec types.SpecTuple
+	Pkg  string
+	Rev  string
+}
+
+// CapacityProviders are a way for packages to be built.
+type CapacityProvider interface {
```

Comment:
Consider dropping Build from the signatures here.  Its known that all of these deal with Builds based on the type.